utils/data_source_manager.py

```
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+import re
+from typing import Any, Dict, List, Optional, Tuple
```

Ruff (0.14.10)

`typing.Dict` is deprecated, use `dict` instead

(UP035)